src/agents/bee/runners/granite/prompts.ts

```diff
@@ -66,21 +67,62 @@ Pay close attention to the tool description to determine if a tool is useful in
 - When the message is unclear, respond with a line starting with 'Final Answer:' followed by a request for additional information needed to solve the problem.
 - When the user wants to chitchat instead, always respond politely.
 
-{{#tools.length}}
-# File Handling
-If a file is provided by the user, make sure you always use the FileRead tool immediately to read the contents of the file.
-{{/tools.length}}
-
-# Date and Time
-The current date and time is contained in the most recent user message.
+# Current Date and Time
+${new Intl.DateTimeFormat(undefined, {
+  dateStyle: "full",
+  timeStyle: "medium",
+}).format(new Date())}.
```

Comment:
This could cause serious bugs because the template instance is re-used and hence if the container/script is running for a little bit it will have outdated information. Next problem relates to date formatting because Intl formats the date based on the current locale which may vary from machine to machine.

Proposed solutions

```ts
const template = new PromptTemplate({
  template: `Current Date: {{date}}`,
  schema: z.object({}),
  functions: {
    date: () =>
      new Intl.DateTimeFormat(undefined, {
        dateStyle: "full",
        timeStyle: "medium",
      }).format(new Date()),
  },
});
console.log(template.render({}));
```

or (better) use the `createdAt` meta property that is part of the message

```
const template = new PromptTemplate({
  template: `# Current Date and Time
{{formatDate}}`,
  schema: z.object({
    createdAt: z.string().datetime().nullish(),
  }),
  functions: {
    formatDate: function () {
      const date = this.createdAt ? new Date(this.createdAt) : new Date();
      return new Intl.DateTimeFormat(undefined, {
        dateStyle: "full",
        timeStyle: "medium",
      }).format(date);
    },
  },
});
```

You would just need to pass the current when rendering the system prompt

Reply:
FYI: Im going with date/time in the system prompt because granite does not pick it up reliably from the User prompt (which is the current approach in Bee). Having it in the system prompt is more reliable. It also allows me to remove the date/time from the user prompt which reduces tokens.

Given the somewhat experimental nature of the granite bee Im ok with the locale issue.

Reply:
I just remembered that you could solve the locale issue by specifying the region as a first parameter (e.g.:"en-US")

Reply:
I believe that by providing an undefined locale it will just default to the runtime's default locale.

Reply:
@Tomas2D Can you recheck this one in particular. Thx.

Reply:
I don't think it is desired. Let's say you run it on a machine in Germany. The result would be `Mittwoch, 11.` Dezember 2024 um 01:00:00` - I don't believe that this is something that the model would be capable of working with.